### ISSUE TITLE
UUID Assignment Data Error 

### ISSUE THREAD

**Issue by sirwolfgang** [nedap/mysql-binuuid-rails#26]
I working on a controller, that functionally works like the snippet below. Mainly that ID is provided by the external source and passed to initialize the object. As I trace it through, it seems to be passing the value correctly all the way down; But goes funny on reflection. 

```ruby
user_params =>
  #<ActionController::Parameters {"first_name"=>"FIRST", "last_name"=>"LAST", "id"=>"92039aae-a540-42e8-85f9-c3cae2d5bc4d"} permitted: true>

@user = User.new(user_params)

#<User:0x00007f2651bac830
 id: "39323033-3961-6165-2d61-3534302d343265382d383566392d633363616532643562633464",
 first_name: "FIRST",
 last_name: "LAST"
>

@attributes['id']
=> #<ActiveModel::Attribute::FromUser:0x00007f26303a0f40
 @name="id",
 @original_attribute=
  #<ActiveModel::Attribute::FromDatabase:0x00007f263044e780 @name="id", @original_attribute=nil, @type=#<ActiveModel::Type::Binary:0x00007f265191a568 @limit=16, @precision=nil, @scale=nil>, @value_before_type_cast=nil>,
 @type=#<MySQLBinUUID::Type:0x000055e2ccb2c2a8 @limit=nil, @precision=nil, @scale=nil>,
 @value="39323033-3961-6165-2d61-3534302d343265382d383566392d633363616532643562633464",
 @value_before_type_cast="92039aae-a540-42e8-85f9-c3cae2d5bc4d">
```

I am not quite sure where the issue is, as I have been unable to discern where the extended value comes from. It doesn't seem to be a completely deserialized value with it haveing `d`s
present. 

As a side note, while tracking this I came across the `assert_valid_value` hook on `Type` which may be a better place to handle certain validations on the content.

**Comment by sirwolfgang** [comment]
Okay, finally figured it out and threw up a patch with #27

**Comment by markoudev** [comment]
Closed through #27
